core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -289,6 +289,30 @@ object HistoryServer extends Logging {
     while(true) { Thread.sleep(Int.MaxValue) }
   }
 
+  /**
+   * Create a security manager.
+   * This includes any fixup of the configurations needed to produce a security manager
+   * capable of starting the History Server.
+   * @param config configuration for the SecurityManager constructor
+   * @return
+   */
+  private[history] def createSecurityManager(config: SparkConf): SecurityManager = {
+    patchSecuritySettings(config)
+    new SecurityManager(config)
+  }
+
+  /**
+   * Fix up the configuration of a spark configuration so that the security manager will
+   * always start.
+   * @param config configuration to be used in a SecurityManager constructor
+   */
+  private def patchSecuritySettings(config: SparkConf): Unit = {
```

Comment:
I've inlined it, which highlights something else. if the debug message is viewed as unimportant, the if() clause can be pulled completely, and the security flag *always* set to false.